ingest/expression_files/expression_files.py

```diff
@@ -93,31 +93,63 @@ def create_gene_model(
         )
 
     @staticmethod
-    def check_unique_cells(cell_names: List, study_id, client):
-        """Checks cell names against database to confirm matrix contains unique
-            cell names.
+    def is_raw_count(study_id, study_file_id, client):
+        COLLECTION_NAME = "study_files"
+        FIELD_NAMES = {"expression_file_info.is_raw_counts": 1, "_id": 0}
+        QUERY = {"_id": study_file_id}
+        # If study files does not have document expression_file_info
+        # field, "is_raw_counts", will not exist.
+        if GeneExpression.has_expression_file_info_doc(study_id, study_file_id, client):
+            query_results = list(client[COLLECTION_NAME].find(QUERY, FIELD_NAMES)).pop()
+            return query_results["expression_file_info"]["is_raw_counts"]
+        else:
+            return False
 
-         Parameters:
-            cell_names (List[str]): List of cell names in matrix
-            study_id (ObjectId): The study id the cell names belong to
-            client: MongoDB client
-        """
-        COLLECTION_NAME = "data_arrays"
-        query = {
+    @staticmethod
+    def query_cells(study_id, client, query_kwargs):
+        QUERY = {
             "$and": [
                 {"linear_data_type": "Study"},
                 {"array_type": "cells"},
                 {"study_id": study_id},
             ],
             "$nor": [{"name": "All Cells"}],
+            **query_kwargs,
         }
         # Returned fields from query results
-        field_names = {"values": 1, "_id": 0}
+        FIELD_NAMES = {"values": 1, "_id": 0}
+        COLLECTION_NAME = "data_arrays"
+
+        return list(client[COLLECTION_NAME].find(QUERY, FIELD_NAMES))
+
+    @staticmethod
+    def check_unique_cells(cell_names: List, study_id, study_file_id, client):
+        """Method checks for unique cells  by matrix type
+
+
+         Parameters:
+            cell_names (List[str]): List of cell names in matrix
+            study_id (ObjectId): The study id the cell names belong to
+            study_file_id (ObjectId): The file id the cell names belong to
+            client: MongoDB client
+        """
+
+        additional_query_kwargs = {}
+        study_files_ids = GeneExpression.get_study_expression_file_ids(
+            study_id, study_file_id, client
+        )
+
```

Comment:
I would pull everything from lines 142-160 into a helper method "get_cell_names_from_study_file_ids" or something like that.  The method can just return an empty array if the study_file_id array is empty

Reply:
7352d5c